app/javascript/react/components/ThresholdConfigurator/ThresholdConfigurator.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import { Thresholds } from "../../types/thresholds";
+import {
+  calculateThumbPercentage,
+  calculateThumbPosition,
+} from "../../utils/thresholdThumbCalculations";
+import * as S from "./ThresholdConfigurator.style";
+
+interface ThresholdsConfiguratorProps {
+  initialThresholds: Thresholds;
+}
+
+interface ThumbPositions extends Omit<Thresholds, "min" | "max"> {}
+
+const ThresholdsConfigurator: React.FC<ThresholdsConfiguratorProps> = ({
+  initialThresholds,
+}) => {
+  const [thresholdValues, setThresholdValues] = useState(initialThresholds);
+  const [thumbPositions, setThumbPositions] = useState<ThumbPositions>(
+    {} as ThumbPositions
+  );
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (sliderRef.current) {
+      setSliderWidth(sliderRef.current.offsetWidth);
+    }
+    const { min, low, middle, high, max } = thresholdValues;
+    const lowThumb = calculateThumbPosition(low, min, max, sliderWidth);
+    const middleThumb = calculateThumbPosition(middle, min, max, sliderWidth);
+    const highThumb = calculateThumbPosition(high, min, max, sliderWidth);
+
+    setThumbPositions({ low: lowThumb, middle: middleThumb, high: highThumb });
+  }, [thresholdValues, sliderWidth]);
+
+  const handleInputChange = (thresholdKey: keyof Thresholds, value: string) => {
+    setThresholdValues({ ...thresholdValues, [thresholdKey]: Number(value) });
+  };
+
+  const handleMouseMove =
+    (thresholdKey: keyof Thresholds, startX: number, startValue: number) =>
+    (moveEvent: globalThis.MouseEvent) => {
+      const dx = moveEvent.clientX - startX;
+      const newPercentage =
+        calculateThumbPercentage(
+          startValue,
+          thresholdValues.min,
+          thresholdValues.max
+        ) +
+        dx / sliderWidth;
+      const newValue = Math.round(
+        thresholdValues.min +
+          newPercentage * (thresholdValues.max - thresholdValues.min)
+      );
+
+      setThresholdValues((prevThresholds) => {
+        return {
+          ...prevThresholds,
+          [thresholdKey]: Math.min(
+            Math.max(newValue, thresholdValues.min),
+            thresholdValues.max
```

Comment:
Maybe we could add some comment to this math, as it is not so clear at first glance. Or the option second, extract something to variable and divide it into smaller? 
Subjective